utilities/transactions/write_prepared_txn_db.h

```diff
@@ -465,6 +465,14 @@ class WritePreparedTxnDB : public PessimisticTransactionDB {
   // Get a dummy snapshot that refers to kMaxSequenceNumber
   Snapshot* GetMaxSnapshot() { return &dummy_max_snapshot_; }
 
+  std::function<bool(TransactionDB*, uint32_t, const Slice&)>
+  GetRollbackDeletionTypeCallback() const {
+    return rollback_deletion_type_callback_;
+  }
```

Comment:
Instead of exposing (a copy of) the function object itself, we could consider adding a method that calls it and returns whether `SingleDelete` should be used for a given CF and key.

Reply:
Will do.

Reply:
Also regarding the overhead of type-erasure of `std::function`, I think it's ok here assuming the rollback ratio is low.

Reply:
Sure, I mean we do need some sort of run-time polymorphism since this will be a customization point for the application